airbyte-workers/src/main/java/io/airbyte/workers/config/ContainerOrchestratorConfigBeanFactory.java

```diff
@@ -42,16 +55,41 @@ public ContainerOrchestratorConfig kubernetesContainerOrchestratorConfig(
                                                                            @Value("${airbyte.container.orchestrator.secret-name}") final String containerOrchestratorSecretName,
                                                                            @Value("${google.application.credentials}") final String googleApplicationCredentials,
                                                                            @Value("${airbyte.worker.job.kube.namespace}") final String namespace,
+                                                                           @Value("${airbyte.metric.client}") final String metricClient,
+                                                                           @Value("${datadog.agent.host}") final String dataDogAgentHost,
+                                                                           @Value("${datadog.agent.port}") final String dataDogStatsdPort,
+                                                                           @Value("${airbyte.metric.should-publish}") final String shouldPublishMetrics,
+                                                                           final FeatureFlags featureFlags,
+                                                                           @Value("${airbyte.container.orchestrator.java-opts}") final String containerOrchestratorJavaOpts,
                                                                            final WorkerEnvironment workerEnvironment) {
     final var kubernetesClient = new DefaultKubernetesClient();
 
     final DocumentStoreClient documentStoreClient = StateClients.create(
         cloudStateStorageConfiguration.orElse(null),
         STATE_STORAGE_PREFIX);
 
+    // Build the map of additional environment variables to be passed to the container orchestrator
+    final Map<String, String> environmentVariables = new HashMap<>();
+    environmentVariables.put(METRIC_CLIENT_ENV_VAR, metricClient);
+    environmentVariables.put(DD_AGENT_HOST_ENV_VAR, dataDogAgentHost);
+    environmentVariables.put(DD_SERVICE_ENV_VAR, "airbyte-container-orchestrator");
+    environmentVariables.put(DD_DOGSTATSD_PORT_ENV_VAR, dataDogStatsdPort);
+    environmentVariables.put(PUBLISH_METRICS_ENV_VAR, shouldPublishMetrics);
+    environmentVariables.put(EnvVariableFeatureFlags.USE_STREAM_CAPABLE_STATE, Boolean.toString(featureFlags.useStreamCapableState()));
+    environmentVariables.put(JAVA_OPTS_ENV_VAR, containerOrchestratorJavaOpts);
+
+    if (System.getenv(DD_ENV_ENV_VAR) != null) {
+      environmentVariables.put(DD_ENV_ENV_VAR, System.getenv(DD_ENV_ENV_VAR));
+    }
+
+    if (System.getenv(DD_VERSION_ENV_VAR) != null) {
+      environmentVariables.put(DD_VERSION_ENV_VAR, System.getenv(DD_VERSION_ENV_VAR));
+    }
```

Comment:
Should this be added to the `application.yml` file?

Reply:
What is going on here is that the env vars are set in the Helm chart, as they need to be available for the DD Java agent to pick up.   We are then copying the ones set on the worker for use in the orchestrator process that will be launched.  We could also use injection but we still have to set it as an env var, so I felt it was just best to leave it as a direct read of the env var.